docs/data/pages.ts

```diff
@@ -505,6 +505,13 @@ const pages: MuiPage[] = [
           { pathname: '/x/react-tree-view/rich-tree-view/focus' },
         ],
       },
+      {
+        pathname: '/x/react-tree-view/common-features',
+        subheader: 'Common features',
+        children: [
+          { pathname: '/x/react-tree-view/tree-item-customization', title: 'Item customization' },
```

Comment:
I took the opportunity to create this page we were talking about
It will also host all the doc about `useTreeItem2` usage and `TreeItem2` slot usage.